Source/WebCore/platform/graphics/displaylists/DisplayListItem.cpp

```diff
@@ -53,14 +53,15 @@ bool isValid(const Item& item)
 template<class T>
 inline static std::optional<RenderingResourceIdentifier> applyFilteredImageBufferItem(GraphicsContext& context, const ResourceHeap& resourceHeap, const T& item)
 {
-    RELEASE_ASSERT(item.sourceImageIdentifier().has_value());
-    auto resourceIdentifier = item.sourceImageIdentifier().value();
-    if (auto* sourceImage = resourceHeap.getImageBuffer(resourceIdentifier)) {
-        FilterResults results;
-        item.apply(context, sourceImage, results);
-        return std::nullopt;
+    ImageBuffer* sourceImage = nullptr;
+    if (auto resourceIdentifier = item.sourceImageIdentifier(); resourceIdentifier.has_value()) {
+        if (sourceImage = resourceHeap.getImageBuffer(resourceIdentifier.value()); !sourceImage)
+            return resourceIdentifier;
     }
-    return resourceIdentifier;
+
+    FilterResults results;
+    item.apply(context, sourceImage, results);
+    return std::nullopt;
```

Comment:
Thanks for removing the `RELEASE_ASSERT`. Apple ports had stopped using `DisplayList` for complex rendering since GPUProcess was enabled by default. We still use the `DisplayList` rendering for very simple cases like recording `TextRun` rendering. So this code has not been tested.

I found it hard to see the conditions in the above two if statement because they come at the end after the semi-colon. I think this code might look clearer:
```
    auto resourceIdentifier = item.sourceImageIdentifier();
    auto sourceImage = resourceIdentifier ? resourceHeap.getImageBuffer(*resourceIdentifier) : nullptr;
    if (UNLIKELY(!sourceImage && resourceIdentifier))
        return resourceIdentifier;

    FilterResults results;
    item.apply(context, sourceImage, results);
    return std::nullopt;
```

Reply:
Thanks @shallawa for the comment -- in fact I added the code a while ago, and it was obviously not well tested. Indeed your snippet looks nicer. I will fix this up in a follow-up PR and add you as reviewer :-)